design/tree.md

```diff
@@ -0,0 +1,166 @@
+
+These coordinate systems transform in the order of '`index(path)` → `IndexTree.TreePos` → `CRDTTree.TreeNodeID` → `CRDTTree.TreePos`'.
+
+<img src="https://github.com/yorkie-team/yorkie/assets/78714820/af339dc7-5c03-4cae-a1cb-f5879bfce3be" />
+
+1. `index` → `IndexTree.TreePos`
+
+The `index` is the coordinate system used by users for local editing. This `index` is received from the user, and is converted to `IndexTree.TreePos`. This `IndexTree.TreePos` represents the physical position within the local tree and is used for actual tree editing.
+
+2. `IndexTree.TreePos` → (`CRDTTree.TreeNodeID`) → `CRDTTree.TreePos`
+
+Next, the obtained `IndexTree.TreePos` is transformed into the logical coordinate system of the distributed tree, represented by `CRDTTree.TreePos`. To achieve this, the given physical position, `IndexTree.TreePos`, is used to find the parent node and left sibling node. Then, a `CRDTTree.TreePos` is created using the unique IDs of the parent node and left sibling node, which are `CRDTTree.TreeNodeID`. This coordinate system is used in subsequent `Tree.Edit` and `Tree.Style` operations.
+
+In the case of remote editing, where the local coordinate system is received from the user in local editing, there is no need for Step 1 since changes are pulled from the server using `ChangePack` to synchronize the changes. Refer to the [document-editing](https://github.com/yorkie-team/yorkie/blob/main/design/document-editing.md) for more details.
+
+**Tree.Edit Logic**
+
+The core process of the `Tree.Edit` operation is as follows:
+
+1. Find `CRDTTree.TreePos` from the given `fromIdx` and `toIdx` (local editing only).
+2. Find the corresponding left sibling node and parent node within the `IndexTree` based on `CRDTTree.TreePos`.
+3. Delete nodes in the range of `fromTreePos` to `toTreePos`.
+4. Insert the given nodes at the appropriate positions (insert operation only).
+
+**[[STEP 1]](https://github.com/yorkie-team/yorkie/blob/fd3b15c7d2c482464b6c8470339bcc497204114e/pkg/document/json/tree.go#L121C1-L128)** Find `CRDTTree.TreePos` from the given `fromIdx` and `toIdx` (local editing only)
+
+In the case of local editing, the given `index`es are converted to `CRDTTree.TreePos`. The detailed process is the same as described in the 'Tree Coordinate System' above.
+
+**[[STEP 2]](https://github.com/yorkie-team/yorkie/blob/fd3b15c7d2c482464b6c8470339bcc497204114e/pkg/document/crdt/tree.go#L572C1-L580C3)** Find the corresponding left sibling node and parent node within the `IndexTree` based on `CRDTTree.TreePos`
+
+2-1. For text nodes, if necessary, split nodes at the appropriate positions to find the left sibling node.
+
+2-2. Determine the sequence of nodes and find the appropriate position. Since `Clone`s[링크] of each client might exist in different states, the `findFloorNode` function is used to find the closest node (lower bound).
```

Comment:
Oh, I missed it.😅 Thank you!

Reply:
I removed the link because I already have the same link above, and can't make a link on a single line in Markdown view!

https://github.com/yorkie-team/yorkie/blob/0293c8dea2e9b9ef2735997bee6d18ca40c14ba2/design/tree.md?plain=1#L86